src/machines/authMachine.ts

```diff
@@ -215,44 +253,74 @@ async function getAndSyncStoredToken(input: {
     return VITE_KITTYCAD_API_TOKEN
   }
 
+  const environmentName = env().VITE_KITTYCAD_BASE_DOMAIN
+
+  // Find possible tokens
   const inputToken = input.token && input.token !== '' ? input.token : ''
   const cookieToken = getCookie(COOKIE_NAME)
-  const localStorageToken = localStorage?.getItem(TOKEN_PERSIST_KEY) || ''
-  const token = inputToken || cookieToken || localStorageToken
+  const fileToken =
+    isDesktop() && environmentName
+      ? await readEnvironmentConfigurationToken(environmentName)
+      : ''
+  const token = inputToken || cookieToken || fileToken
 
+  // Log what tokens we found
   console.log('Token used for authentication')
   console.table([
     ['persisted token', !!inputToken],
     ['cookie', !!cookieToken],
-    ['local storage', !!localStorageToken],
     ['api token', !!VITE_KITTYCAD_API_TOKEN],
+    ['file token', !!fileToken],
   ])
+
+  // If you found a token
   if (token) {
+    // Write it to disk to sync it for desktop!
     if (isDesktop()) {
       // has just logged in, update storage
-      localStorage.setItem(TOKEN_PERSIST_KEY, token)
-      await writeTokenFile(token)
+      if (environmentName)
+        await writeEnvironmentConfigurationToken(environmentName, token)
     }
     return token
   }
+
+  // If you are web and you made it this far, you do not get a token
   if (!isDesktop()) return ''
-  const fileToken = isDesktop() ? await readTokenFile() : ''
-  // prefer other above, but file will ensure login persists after app updates
+
   if (!fileToken) return ''
-  // has token in file, update localStorage
-  localStorage.setItem(TOKEN_PERSIST_KEY, fileToken)
+  // default desktop login workflow to always read from disk, file will ensure login persists after app updates
   return fileToken
 }
 
+/**
+ * Logout function that will do a default logout within the AuthMachine
+ */
 async function logout() {
+  return logoutEnvironment()
+}
+
+/**
+ * Logout function that will do a specific environment logout if environment name is passed in
+ */
+async function logoutEnvironment(requestedEnvironmentName?: string) {
+  // TODO: 7/10/2025 Remove this months from now, we want to clear the localStorage of the key.
   localStorage.removeItem(TOKEN_PERSIST_KEY)
   if (isDesktop()) {
     try {
-      let token = await readTokenFile()
+      const environmentName =
+        requestedEnvironmentName || env().VITE_KITTYCAD_BASE_DOMAIN
+      let token = ''
+      if (environmentName) {
+        token = await readEnvironmentConfigurationToken(environmentName)
+      }
+      if (!environmentName) {
+        return new Error('Unable to logout, cannot find environment')
+      }
 
+      const url = environmentName
       if (token) {
         try {
-          await fetch(withAPIBaseURL('/oauth2/token/revoke'), {
+          await fetch(url + '/oauth2/token/revoke', {
```

Comment:
This variable aliasing feels a bit off to me:

`VITE_KITTYCAD_BASE_DOMAIN` -> `environmentName` -> `url`


Is it the name of an environment or the domain? It's certainly not a complete URL with schema -- does `fetch()` required that?

Reply:
Well I got into the pickle of everything was an `environment` or `environmentName` but I left some of the variable names the same. So all of these are domains now. Trying to clean this block of code up.